--- a/Graphics/GraphicsEngineVulkan/src/VulkanUtilities/Instance.cpp
+++ b/Graphics/GraphicsEngineVulkan/src/VulkanUtilities/Instance.cpp
@@ -384,13 +384,16 @@
         InstanceExtensions.push_back(VK_KHR_ANDROID_SURFACE_EXTENSION_NAME);
 #endif
 #if defined(VK_USE_PLATFORM_WAYLAND_KHR)
-        InstanceExtensions.push_back(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
+        if (IsExtensionAvailable(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME))
+            InstanceExtensions.push_back(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
 #endif
 #if defined(VK_USE_PLATFORM_XLIB_KHR)
-        InstanceExtensions.push_back(VK_KHR_XLIB_SURFACE_EXTENSION_NAME);
+        if (IsExtensionAvailable(VK_KHR_XLIB_SURFACE_EXTENSION_NAME))
+            InstanceExtensions.push_back(VK_KHR_XLIB_SURFACE_EXTENSION_NAME);
 #endif
 #if defined(VK_USE_PLATFORM_XCB_KHR)
-        InstanceExtensions.push_back(VK_KHR_XCB_SURFACE_EXTENSION_NAME);
+        if (IsExtensionAvailable(VK_KHR_XCB_SURFACE_EXTENSION_NAME))
+            InstanceExtensions.push_back(VK_KHR_XCB_SURFACE_EXTENSION_NAME);
 #endif
 #if defined(VK_USE_PLATFORM_METAL_EXT)
         InstanceExtensions.push_back(VK_EXT_METAL_SURFACE_EXTENSION_NAME);
--- a/Graphics/GraphicsEngineVulkan/src/SwapChainVkImpl.cpp
+++ b/Graphics/GraphicsEngineVulkan/src/SwapChainVkImpl.cpp
@@ -107,6 +107,9 @@
 #    if defined(VK_USE_PLATFORM_XCB_KHR)
     if (m_Window.pXCBConnection != nullptr && m_Window.WindowId != 0)
     {
+        if (!m_Instance->IsExtensionEnabled(VK_KHR_XCB_SURFACE_EXTENSION_NAME))
+            LOG_ERROR_AND_THROW("Requested XCB surface extension is not enabled");
+
         VkXcbSurfaceCreateInfoKHR surfaceCreateInfo{};
         surfaceCreateInfo.sType      = VK_STRUCTURE_TYPE_XCB_SURFACE_CREATE_INFO_KHR;
         surfaceCreateInfo.connection = static_cast<xcb_connection_t*>(m_Window.pXCBConnection);
@@ -119,6 +122,9 @@
 #    if defined(VK_USE_PLATFORM_XLIB_KHR)
     if ((m_Window.pDisplay != nullptr && m_Window.WindowId != 0) && m_VkSurface == VK_NULL_HANDLE)
     {
+        if (!m_Instance->IsExtensionEnabled(VK_KHR_XLIB_SURFACE_EXTENSION_NAME))
+            LOG_ERROR_AND_THROW("Requested XLIB surface extension is not enabled");
+
         VkXlibSurfaceCreateInfoKHR surfaceCreateInfo{};
         surfaceCreateInfo.sType  = VK_STRUCTURE_TYPE_XLIB_SURFACE_CREATE_INFO_KHR;
         surfaceCreateInfo.dpy    = static_cast<Display*>(m_Window.pDisplay);
@@ -131,6 +137,9 @@
 #    if defined(VK_USE_PLATFORM_WAYLAND_KHR)
     if ((m_Window.pDisplay != nullptr && m_Window.pWaylandSurface != nullptr) && m_VkSurface == VK_NULL_HANDLE)
     {
+        if (!m_Instance->IsExtensionEnabled(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME))
+            LOG_ERROR_AND_THROW("Requested WAYLAND surface extension is not enabled");
+
         VkWaylandSurfaceCreateInfoKHR surfaceCreateInfo{};
         surfaceCreateInfo.sType   = VK_STRUCTURE_TYPE_WAYLAND_SURFACE_CREATE_INFO_KHR;
         surfaceCreateInfo.display = static_cast<struct wl_display*>(m_Window.pDisplay);
